common/src/View/MapDocument.cpp

```diff
@@ -1568,7 +1568,7 @@ namespace TrenchBroom {
             auto selectedLinkedGroupIds = kdl::vec_transform(selectedGroupsWithLinkGroupIds, [](const auto& g) { return *g->group().linkedGroupId(); });
```

Comment:
Would it help to split this up so `separateSelectedLinkedGroups` is just responsible for examining the selected groups and figuring out what to do with the different subsets, then having the actual work split out into these functions?
```
/** Converts each of the given groups into a non-linked group. They can be part of different link sets. */
MapDocument::unlinkGroups(const std::vector<Model::GroupNode*>&)
/** Relink the given groups into a new link set, separating them from the others in the link set. They must all be in the same link set initially. */
MapDocument::relinkGroups(const std::vector<Model::GroupNode*>&)
```

Reply:
Somethink like 8a4f25934?